crates/indexer/src/geyser/accounts/graph.rs

```diff
@@ -22,7 +28,51 @@ pub(crate) async fn process(client: &Client, key: Pubkey, account_data: Connecti
                 .on_conflict(graph_connections::address)
                 .do_update()
                 .set(&row)
-                .execute(db)
+                .execute(db)?;
+
+            db.build_transaction().read_write().run(|| {
+                let follow_event_exists = select(exists(
+                    follow_events::table
+                        .filter(follow_events::graph_connection_address.eq(row.address.clone())),
+                ))
+                .get_result::<bool>(db);
+
+                if Ok(true) == follow_event_exists {
+                    return Result::<_>::Ok(());
+                }
```

Comment:
Since there is a unique constraint on each event type table we could drop the select and let the transaction rollback that is already occurring when 2 consumer jobs processing the same account change for the first time.